File: backend/app/schemas/location.py
from typing import Optional
from pydantic import BaseModel, Field


class CreateLocationRequest(BaseModel):
    latitude: float = Field(..., ge=-90, le=90)
    longitude: float = Field(..., ge=-180, le=180)
    location_type: str
    formatted_address: Optional[str] = None
    city: Optional[str] = None
    state_region: Optional[str] = None
    country: Optional[str] = None
    postal_code: Optional[str] = None

File: frontend/src/types/location.ts
export type LocationType =
  | "customer_address"
  | "tailor_shop"
  | "vendor_warehouse"
  | "delivery_dropoff"
  | "delivery_pickup";

export interface GeoLocation {
  id: string;
  latitude: number;
  longitude: number;
  location_type: LocationType;
  formatted_address?: string | null;
  city?: string | null;
  state_region?: string | null;
  country?: string | null;
  postal_code?: string | null;
}

export interface Address {
  id: string;
  label?: string | null;
  recipient_name?: string | null;
  recipient_phone?: string | null;
  is_default: boolean;
}
